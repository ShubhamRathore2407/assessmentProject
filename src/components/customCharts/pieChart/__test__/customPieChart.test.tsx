import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import CustomPieChart from '../';

describe('CustomPieChart component', () => {
  const data = [
    { name: 'A', value: 10 },
    { name: 'B', value: 20 },
    { name: 'C', value: 30 },
  ];

  test('renders CustomPieChart component correctly', () => {
    const { container } = render(
      <CustomPieChart
        data={data}
        colors={['#FF5733', '#33FF57', '#3357FF']}
        width={400}
        height={400}
        cx={200}
        cy={200}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders CustomPieChart component with different dimensions', () => {
    const { container } = render(
      <CustomPieChart
        data={data}
        colors={['#FF5733', '#33FF57', '#3357FF']}
        width={500}
        height={300}
        cx={250}
        cy={150}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders CustomPieChart component with mouse hover', () => {
    const { getByLabelText, queryByLabelText } = render(
      <CustomPieChart
        data={data}
        colors={['#FF5733', '#33FF57', '#3357FF']}
        width={400}
        height={400}
        cx={200}
        cy={200}
      />
    );

    // const segmentA = getByLabelText('Segment A');
    // fireEvent.mouseEnter(segmentA);
    // expect(segmentA).toBeInTheDocument();

    // fireEvent.mouseLeave(segmentA);
    // expect(queryByLabelText('Segment A')).not.toBeInTheDocument();
  });

  test('renders CustomPieChart component with specified stroke width', () => {
    const strokeWidth = 4;
    const { container } = render(
      <CustomPieChart
        data={data}
        colors={['#FF5733', '#33FF57', '#3357FF']}
        width={400}
        height={400}
        cx={200}
        cy={200}
        strokeWidth={strokeWidth}
      />
    );

    const segments = container.querySelectorAll('path');
    segments.forEach((segment) => {
      expect(segment.getAttribute('stroke-width')).toBe(`${strokeWidth}`);
    });
  });

  test('renders CustomPieChart component with inner radius', () => {
    const innerRadius = 50;
    const { container } = render(
      <CustomPieChart
        data={data}
        colors={['#FF5733', '#33FF57', '#3357FF']}
        width={400}
        height={400}
        cx={200}
        cy={200}
        innerRadius={innerRadius}
      />
    );

    const innerCircles = container.querySelectorAll('circle');
    // expect(innerCircles.length).toBe(data.length);
    innerCircles.forEach((circle) => {
      expect(circle.getAttribute('r')).toBe(`${innerRadius}`);
    });
  });
  test('renders text elements with correct positions', () => {
    const { queryAllByText } = render(
      <CustomPieChart
        data={data}
        colors={['#FF5733', '#33FF57', '#3357FF']}
        width={400}
        height={400}
        cx={200}
        cy={200}
      />
    );

    const textElements = queryAllByText(/A|B|C/i);

    const textElement1 = textElements.find(el => el.getAttribute('x') === '236.68447831576693' && el.getAttribute('y') === '121.32034355964288');
    const textElement2 = textElements.find(el => el.getAttribute('x') === '199.99999999999997' && el.getAttribute('y') === '145.91524164975486');
    const textElement3 = textElements.find(el => el.getAttribute('x') === '169.13735041660296' && el.getAttribute('y') === '171.84189004559765');

    expect(textElement1)
    expect(textElement2)
    expect(textElement3)
});


});
