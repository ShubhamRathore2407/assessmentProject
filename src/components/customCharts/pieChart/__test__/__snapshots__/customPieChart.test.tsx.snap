// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomPieChart component renders CustomPieChart component correctly 1`] = `
<div
  class="recharts-wrapper"
  role="region"
  style="position: relative; cursor: default; width: 400px; height: 400px;"
>
  <svg
    class="recharts-surface"
    cx="50%"
    cy="50%"
    height="400"
    style="width: 100%; height: 100%;"
    viewBox="0 0 400 400"
    width="400"
  >
    <title />
    <desc />
    <defs>
      <clippath
        id="recharts1-clip"
      >
        <rect
          height="390"
          width="390"
          x="5"
          y="5"
        />
      </clippath>
    </defs>
    <g
      class="recharts-layer recharts-pie"
      tabindex="0"
    >
      <g
        class="recharts-layer"
      />
    </g>
  </svg>
</div>
`;

exports[`CustomPieChart component renders CustomPieChart component with different dimensions 1`] = `
<div
  class="recharts-wrapper"
  role="region"
  style="position: relative; cursor: default; width: 500px; height: 300px;"
>
  <svg
    class="recharts-surface"
    cx="50%"
    cy="50%"
    height="300"
    style="width: 100%; height: 100%;"
    viewBox="0 0 500 300"
    width="500"
  >
    <title />
    <desc />
    <defs>
      <clippath
        id="recharts3-clip"
      >
        <rect
          height="290"
          width="490"
          x="5"
          y="5"
        />
      </clippath>
    </defs>
    <g
      class="recharts-layer recharts-pie"
      tabindex="0"
    >
      <g
        class="recharts-layer"
      />
    </g>
  </svg>
</div>
`;
